eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -285,24 +285,24 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
           context.system.eventStream.publish(ShortChannelIdAssigned(self, normal.channelId, normal.channelUpdate.shortChannelId, None))
 
           // we rebuild a new channel_update with values from the configuration because they may have changed while eclair was down
-          // NB: we don't update the routing fees, because we don't want to overwrite manual changes made with CMD_UPDATE_RELAY_FEE
-          // Since CMD_UPDATE_RELAY_FEE is handled even when being offline, that's the preferred solution to update routing fees
+          val fees = getRelayFees(nodeParams, remoteNodeId, data.commitments)
           val candidateChannelUpdate = Announcements.makeChannelUpdate(
             nodeParams.chainHash,
             nodeParams.privateKey,
             remoteNodeId,
             normal.channelUpdate.shortChannelId,
             nodeParams.expiryDelta,
             normal.commitments.remoteParams.htlcMinimum,
-            normal.channelUpdate.feeBaseMsat,
-            normal.channelUpdate.feeProportionalMillionths,
+            fees.feeBase,
+            fees.feeProportionalMillionths,
             normal.commitments.capacity.toMilliSatoshi,
             enable = Announcements.isEnabled(normal.channelUpdate.channelFlags))
           val channelUpdate1 = if (Announcements.areSame(candidateChannelUpdate, normal.channelUpdate)) {
             // if there was no configuration change we keep the existing channel update
             normal.channelUpdate
           } else {
             log.info("refreshing channel_update due to configuration changes old={} new={}", normal.channelUpdate, candidateChannelUpdate)
+            nodeParams.db.audit.addChannelUpdate(data.channelId, candidateChannelUpdate)
```

Comment:
The usual pattern is to rely on events to update the `audit` db, and make the calls from within the `DbEventHandler`. Main reason was to minimize performance impacts. Here there is no real perf concern since relay fees are not updated often (except if we update all channels at once), but it's still better to have as little blocking db calls as possible in the channel, and it would still be better for consistency.

In any case I would factor it in the `onTransition`. We actually already emit a `LocalChannelUpdate` event (see https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala#L1881-L1904). However the event is also emitted if we simply enable/disable the channel. The `DbEventHandler` could listen to that event and only store the update in the audit db when the channel is enabled.

This would also mean that if the channel is being created, we would store the event in the audit db when the channel is ready and not before. Same if the channel is offline. I think that's ok.

Reply:
Done